spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/LocallyConnected2D.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn.keras
+
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, DataFormat, TensorModule}
+import com.intel.analytics.bigdl.optim.Regularizer
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Shape
+
+import scala.reflect.ClassTag
+
+/**
+  * Locally-connected layer for 2D inputs.
+  * The LocallyConnected2D layer works similarly to the SpatialConvolution layer,
+  * except that weights are unshared, that is, a different set of filters
+  * is applied at each different patch of the input.
+  * When using this layer as the first layer in a model, you need to provide the argument
+  * inputShape (a Single Shape, does not include the batch dimension),
+  * e.g. inputShape=(3, 128, 128) for 128x128 RGB pictures.
+  *
+  * @param nbFilter Number of convolution filters to use.
+  * @param nbRow Number of rows in the convolution kernel.
+  * @param nbCol Number of columns in the convolution kernel.
+  * @param activation activation function to use.Default is null.
+  *                   You can also pass in corresponding string representations such as 'relu'
+  *                   or 'sigmoid', etc. for simple activations in the factory method.
+  * @param borderMode Either 'valid' or 'same'. Default is 'valid'.
+  * @param subsample Int array of length 2. The step of the convolution in the height and
+  *                  width dimension. Also called strides elsewhere. Default is (1, 1).
+  * @param wRegularizer An instance of [[Regularizer]], (eg. L1 or L2 regularization),
+  *                     applied to the input weights matrices. Default is null.
+  * @param bRegularizer An instance of [[Regularizer]], applied to the bias. Default is null.
+  * @param format Format of the input data. Either DataFormat.NCHW or DataFormat.NHWC. Default is NCHW.
+  * @param bias Whether to include a bias (i.e. make the layer affine rather than linear).
+  *             Default is true.
+  * @tparam T The numeric type of parameter(e.g. weight, bias). Only support float/double now
+  */
+class LocallyConnected2D[T: ClassTag](
+   val nbFilter: Int,
+   val nbRow: Int,
+   val nbCol: Int,
+   val activation: TensorModule[T] = null,
```

Comment:
Need to change to AbstractModule[Tensor[T], Tensor[T], T], refer to Convolution2D.